### PR TITLE
fix: change default mongodb replica set to null

### PR DESCRIPTION
**Pull request by kaustavb12**
Configuration Pull Request
---

Make sure that the following steps are done before merging:

  - [ ] A SRE team member has approved the PR if it is code shared across multiple services and you don't own all of the services.
  - [ ] Are you adding any new default values that need to be overridden when this change goes live? If so:
    - [ ] Update the appropriate internal repo (be sure to update for all our environments)
    - [ ] If you are updating a secure value rather than an internal one, file a SRE ticket with details.
    - [x] Add an entry to the CHANGELOG.
  - [ ] If you are making a complicated change, have you performed the proper testing specified on the [Ops Ansible Testing Checklist](https://openedx.atlassian.net/wiki/spaces/SRE/pages/28967861/Ops+Ansible+Testing+Checklist)?  Adding a new variable does not require the full list (although testing on a sandbox is a great idea to ensure it links with your downstream code changes).
  - [ ] Think about how this change will affect Open edX operators.  Have you updated the wiki page for the next Open edX release?

## Description

This PR changes the default value of `EDXAPP_MONGO_REPLICA_SET` from empty string (`""`) to null.

This is required due to unpinnig of pymongo via [edx-platform#30569](https://github.com/openedx/edx-platform/pull/30569) 
As mentioned in that PR, `pymongo >= 3.11` only accepts `None` as replica set if standalone MongoDB instance is used.
Empty string is not accepted as a valid value for replica set.

## Supporting information

Link to other information about the change, such as Jira issues, GitHub issues, or Discourse discussions.
Be sure to check they are publicly readable, or if not, repeat the information here.

## Testing instructions

1. Spinup an instance using a standalone MongoDB instance without passing `EDXAPP_MONGO_REPLICA_SET`
2. Ensure the instance spins up correctly, and all migrations run correctly.

## Deadline

"None" if there's no rush, or provide a specific date or event (and reason) if there is one.

## Other information

Private Ref: [BB-6773](https://tasks.opencraft.com/browse/BB-6773)

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @kaustavb12!  Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

File: playbooks/roles/edxapp/defaults/main.yml [hunk 1]
```diff
@@ -74,7 +74,7 @@ EDXAPP_MONGO_PORT: 27017
 EDXAPP_MONGO_USER: 'edxapp'
 EDXAPP_MONGO_DB_NAME: 'edxapp'
 EDXAPP_MONGO_USE_SSL: False
-EDXAPP_MONGO_REPLICA_SET: ''
+EDXAPP_MONGO_REPLICA_SET: null
```
Comment (navinkarkera) [inline playbooks/roles/edxapp/defaults/main.yml:77]: @kaustavb12 Why do we have `!!null` is some places like https://github.com/openedx/configuration/blob/1ad3ecf3e73f28f8077b5aa8577ddc64b34dcbf6/playbooks/roles/edxapp/defaults/main.yml#L19

I am not sure about the difference here.

**Comment by navinkarkera** [comment]
:+1:
- [x] I tested this: checked deployment of nutmeg instance
- [x] I read through the code
- [ ] I checked for accessibility issues
- [ ] Includes documentation
- [ ] I made sure any change in configuration variables is reflected in the corresponding client's `configuration-secure` repository.

File: playbooks/roles/edxapp/defaults/main.yml (hunk 1 above)
Comment (kaustavb12) [reply playbooks/roles/edxapp/defaults/main.yml:77]: @navinkarkera 

`!!` notation is used to explicitly denote the [type of an item](https://yaml.org/spec/1.2.2/#tags). In this case `!!null` explicitly denotes the [null type](https://yaml.org/type/null.html). However, it seems it in not required to use this notation for `null` as the tag type is [resolved automatically](https://yaml.org/spec/1.2-old/spec.html#id2805071).

Comment (navinkarkera) [reply playbooks/roles/edxapp/defaults/main.yml:77]: @kaustavb12 Thanks for the clarification!

**Comment by mphilbrick211** [comment]
Hi @kaustavb12 and @navinkarkera! Is this PR ready for review?

**Comment by kaustavb12** [comment]
Hi @mphilbrick211 

Thanks for checking. This is ready for review.

**Comment by mphilbrick211** [comment]
@adzuci flagging for your review!

**Comment by mphilbrick211** [comment]
Hi @adzuci! Friendly ping on this :)

**Comment by mphilbrick211** [comment]
@adzuci - hi there - do you have an ETA on when you can review/merge this PR? Thanks!

**Comment by mphilbrick211** [comment]
Hi @adzuci - following up to see if you have an update on this?

**Review by adzuci — APPROVED** [review]

**Comment by mphilbrick211** [comment]
Hi @kaustavb12! Would you mind resolving the branch conflicts on this PR? 

CC: @adzuci for merging after.

**Comment by kaustavb12** [comment]
@mphilbrick211 Conflicts resolved.

@adzuci Thanks for the review and approval

**Comment by openedx-webhooks** [comment]
@kaustavb12 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/configuration&entry.867055334=https://github.com/openedx/configuration/pull/6831&entry.1484655318=https://github.com/kaustavb12&entry.752974735=2022-11-11+02:39&entry.1917517419=2023-04-10+08:41&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**